calico_node/Makefile

```diff
@@ -133,13 +133,15 @@ dist/calicoctl-v1.0.2:
 # variables.  These are used for the STs.
 dist/calicoctl:
 	-docker rm -f calicoctl
+	docker pull $(CTL_CONTAINER_NAME)
```

Comment:
This is probably good to do so we don't end up with unexpected versions in the builds, but has the unfortunate side effect of requiring the images to be pushed, making it harder to do a local-only build.

Reply:
I thought if you pulled an image which was already on your machine it would simply say "image is up to date"?

Reply:
@lwr20 only if the image is actually the same as the remote, and a remote image exists.

For example, I used to be able to build a custom Felix and tag it `caseydavenport/felix:my-special-build` and then point this Makefile at that version and it will work.

With this change, I need to also push that image to a registry else the make target will fail with:

```
Error response from daemon: manifest for caseydavenport/felix:my-spcial-build not found
```

Reply:
@caseydavenport @lwr20 I've posted a possible solution at #1324. WDYT?

Alternatively, we could say that people should set up a local registry, for this use case.  (But I don't know if that is a simple thing to do, and it would probably still need some Makefile coding changes, because I think some container names are not yet overridable.)